service/src/main.rs

```diff
@@ -782,3 +789,33 @@ fn bootstrap_funds_from_alice(
 	api.signer = signer_orig;
 	Ok(())
 }
+
+/// In case this is the first validateer on the specified parentchain, we need to trigger
+/// the parentchain block import up until the block where we have registered ourselves.
+fn bootstrap_parentchain_blocks_for_primary_validateer<E: EnclaveBase + TeerexApi + Sidechain>(
+	node_api: &Api<sr25519::Pair, WsRpcClient>,
+	enclave_api: Arc<E>,
+	last_synced_header: &Header,
+	registry_block_hash: Option<Hash>,
+) -> Result<Header, Error> {
+	let registry_header: Header =
+		node_api.get_header(registry_block_hash)?.ok_or(Error::EmptyValue)?;
+	let mut last_synced_header = last_synced_header.clone();
+	let enclave_count_of_previous_block =
+		node_api.enclave_count(Some(*registry_header.parent_hash()))?;
+
+	if enclave_count_of_previous_block == 0 {
+		info!(
+			"We're the first validateer to be registered, syncing parentchain blocks until the one we have registered ourselves on."
+		);
+		let parentchain_block_syncer =
+			ParentchainBlockSyncer::new(node_api.clone(), enclave_api.clone());
+		while last_synced_header.number() < registry_header.number() {
+			// Otherwise we're spamming syncing messages.
+			thread::sleep(Duration::from_secs(1));
+			last_synced_header = parentchain_block_syncer.sync_parentchain(last_synced_header);
```

Comment:
What's the problem here with spamming messages? Let's say if we have already 100 blocks, this will prolong the syncing process for nothing.

Reply:
It won't, the highest dealy is 1s. `sync_parentchain()` syncs up to all block it sees. But agreed - this is outdated anyway. I'll remove it.

Reply:
Ah ok, it see, my scenario will not take place because we synced before already.

Reply:
Removed